From 83c591fa654874ddb7cb51749905774c1f563b42 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Sun, 8 Aug 2021 12:17:37 +0200
Subject: [PATCH] Remove "using namespace std" from global namespace

Any file which includes pfsutils.cpp implicitly uses the std namespace.
This causes problems for pfsinyuv.cpp, which defines a clamp functions,
while C++17 also has std::clamp.
---
 src/fileformat/pfsinyuv.cpp  | 2 +-
 src/fileformat/pfsoutyuv.cpp | 2 +-
 src/pfs/pfsutils.cpp         | 3 +--
 3 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/fileformat/pfsinyuv.cpp b/src/fileformat/pfsinyuv.cpp
index 5244099..e8d0e25 100644
--- a/src/fileformat/pfsinyuv.cpp
+++ b/src/fileformat/pfsinyuv.cpp
@@ -125,7 +125,7 @@ bool parseFileName(const char * fileNameIn, int * width,  int * height, int * bi
     }
     tokens.push_back(fileName);
 
-    for (std::vector<string>::iterator tok = tokens.begin() + 1; tok != tokens.end(); ++tok) {
+    for (std::vector<std::string>::iterator tok = tokens.begin() + 1; tok != tokens.end(); ++tok) {
       
       if (contains(*tok, "x")){
         sscanf(tok->c_str(), "%dx%d", width, height);
diff --git a/src/fileformat/pfsoutyuv.cpp b/src/fileformat/pfsoutyuv.cpp
index 83b3c24..233a40e 100644
--- a/src/fileformat/pfsoutyuv.cpp
+++ b/src/fileformat/pfsoutyuv.cpp
@@ -364,7 +364,7 @@ void writeFrames( int argc, char* argv[] )
   }
 
   if(fpsTag != NULL){
-    VERBOSE_STR << "FPS Tag: " << fpsTag << endl;
+    VERBOSE_STR << "FPS Tag: " << fpsTag << std::endl;
     fps = atoi(fpsTag);
   }
 
diff --git a/src/pfs/pfsutils.cpp b/src/pfs/pfsutils.cpp
index 4a3a56e..41ba964 100644
--- a/src/pfs/pfsutils.cpp
+++ b/src/pfs/pfsutils.cpp
@@ -36,10 +36,9 @@
 
 #define MAX_FRAME 99999
 
-using namespace std;
-
 namespace pfs
 {
+using namespace std;
 
 string getColorspaceString(ColorSpace cs){
   switch(cs){
-- 
2.32.0

